### PR TITLE
Remove internal package, use single main.go

### PR DESCRIPTION
**Pull request by vincepri**

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -1,66 +1,48 @@
 linters:
   disable-all: true
-  # Enable specific linter
-  # https://golangci-lint.run/usage/linters/#enabled-by-default-linters
   enable:
-    # Default linters
```
Comment (JoelSpeed) [inline .golangci.yaml:6]: What's the motivation behind dropping the linters? I copied this from other projects this team owns

File: main.go
Comment (JoelSpeed) [inline main.go:1]: Can we put this into `cmd/cluster-api-provider-openshift/main.go`, that would then be consistent with other projects the team owns

**Review by JoelSpeed — COMMENTED** [review]
Just a couple of points about being consistent with other projects this teams own

File: .golangci.yaml (hunk 1 above)
Comment (vincepri) [reply .golangci.yaml:6]: Some of them were deprecated/archived. I got the latest list from the CAPI project upstream

File: main.go
Comment (vincepri) [reply main.go:1]: Done

File: .golangci.yaml (hunk 1 above)
Comment (vincepri) [reply .golangci.yaml:6]: The diff is a bit strange due to the ordering of the list

Comment (JoelSpeed) [reply .golangci.yaml:6]: These linter changes are removing
```
- contextcheck
- cyclop
- dupl
- errname
- errorlint
- exhaustive
- forcetypeassert
- funlen
- gochecknoglobals
- gocognit
- gocyclo
- goerr113
- goheader
- ifshort
- makezero
- nestif
- nilnil
- nlreturn
- tagliatelle
- tenv
- wastedassign
- wrapcheck
- wsl
```
and adding
```
- asasalint
- containedctx
- dupword
- errchkjson
- ginkgolinter
- nosprintfhostport
- rowserrcheck
- thelper
- usestdlibvars
```
I would appreciate if we can keep
```
- contextcheck
- goerr113
- wastedassign
- wrapcheck
- wsl
```
To be honest I'm surprised a few of those aren't already in the CAPI list

**Review by JoelSpeed — APPROVED** [review]
/approve
/lgtm

**Pull request merged**